Test/1027

irvaniamirali opened the pull request.

babakjahan committed:
redundant type conversion fixed

babakjahan committed:
test structure for e2e updated

irvaniamirali committed:
docs: add README_PYTEST with E2E test execution instructions

irvaniamirali committed:
refactor(user): use concrete UserRepository in dependency instead of interface

irvaniamirali committed:
fix(tests/e2e): update API_URL_V1 path to match /v1 endpoint

irvaniamirali committed:
test(e2e): add fixture unique_user_data for generating unique test users

irvaniamirali committed:
fix(tests/e2e): update server status test to use lowercase 'status' key

irvaniamirali committed:
test(e2e): add user registration E2E tests for success, duplicate, invalid email, and weak password

irvaniamirali committed:
test(e2e): register user fixtures in conftest.py

irvaniamirali committed:
feat(logger): add TEST_SERVICE to ServiceName enum

irvaniamirali committed:
fix(fixtures): update logger import path and comment out logger fixture

irvaniamirali committed:
refactor(api): remove unused delete_user_by_email function

irvaniamirali committed:
Remove username field from user fixture

babakjahan commented on tests/e2e/fixtures/users.py:
```diff
@@ -0,0 +1,13 @@
+@pytest.fixture
+def unique_user_data():
+    unique_suffix = str(uuid.uuid4())[:8]
+    user_data = {
```
not hard code user here! 
save in data folder

babakjahan commented on tests/e2e/test_cases/test_users.py:
```diff
@@ -0,0 +1,46 @@
+
+
+def test_create_user_weak_password(api_request_context):
+    user_data = {
```
do not hard code user here!
save in data folder

babakjahan commented on tests/e2e/test_cases/test_users.py:
```diff
@@ -0,0 +1,46 @@
+
+def test_create_user_invalid_email(api_request_context):
+    user_data = {
+        "email": "not-an-email",
```
do not hard code user here!
save in data folder

irvaniamirali committed:
test: add sample users with normal and weak passwords

irvaniamirali committed:
test(fixtures): refactor user fixtures to use shared USERS data

irvaniamirali committed:
test(fixtures): add invalid_email user and fixture for testing email validation

Commit:
update test structures

Commit:
update pytest and run paths

Commit:
update preview files for test case

irvaniamirali committed:
Add E2E auth tests and fixtures

- Added `registered_user` fixture to `tests/e2e/fixtures/users.py` for creating a registered user.
- Created `tests/e2e/routes_api_v1/auth.py` with `post_login` helper.
- Implemented login E2E tests:
  - test login for disabled users
  - test login for unverified users
  - test login with wrong password
  - test login with correct credentials
- Updated import paths and cleaned up `1_register_user_tests.py` and `10_login_user_tests.py`.

irvaniamirali committed:
Add E2E category fixtures, routes, and CRUD tests

- Added category fixture to `tests/e2e/fixtures/category.py` for reusable test data.
- Added test data definitions in `tests/e2e/data/category.py`.
- Created `tests/e2e/routes_api_v1/category.py` with API helper functions.
- Updated `20_category_crud_tests.py` with E2E tests for category CRUD operations.

irvaniamirali committed:
Add E2E product fixtures, routes, and CRUD tests

- Added product test data in `tests/e2e/data/product.py`.
- Created API helper functions for products in `tests/e2e/routes_api_v1/product.py`.
- Updated `30_product_crud_tests.py` with E2E tests for product CRUD operations.

irvaniamirali committed:
Register category fixture in E2E conftest

- Added "tests.e2e.fixtures.category" to pytest_plugins for global availability in E2E tests.
